omod/src/main/java/org/openmrs/module/bedmanagement/rest/resource/BedDetailsResource.java

```diff
@@ -51,13 +51,18 @@ public BedDetails getByUniqueId(String id) {
 	}
 	
 	@Override
-	protected void delete(BedDetails bedDetails, String reason, RequestContext requestContext) throws ResponseException {
+	public void delete(String id, String reason, RequestContext requestContext) throws ResponseException {
 		String patientUuid = requestContext.getParameter("patientUuid");
 		BedManagementService bedManagementService = (BedManagementService) Context
 		        .getModuleOpenmrsServices(BedManagementService.class.getName()).get(0);
 		bedManagementService.unAssignPatientFromBed(Context.getPatientService().getPatientByUuid(patientUuid));
 	}
 	
+	@Override
+	protected void delete(BedDetails bedDetails, String s, RequestContext requestContext) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException("delete of bed not supported");
```

Comment:
Unfortunately, this is an abstract class on the super classes, meaning I was forced to provide some sort of implementation here.

For what it's worth, when you make a DELETE request is it now the earlier method that is called (though I'm not sure why)

Reply:
Why don't you just support this by doing something like

`delete(bedDetails.getBed().getId().toString(), ...)`

because your message of `delete of bed not supported` isn't really correct...

Reply:
Yeah, I didn't love my message, but I don't want to actually implement it because this is the method that fails due to the Hibernate exception.  Ideally I'd just remove this method entirely, but it's an abstract class on a super class.  
I'll change the error message to just "not supported" and add a comment linking to the ticket.

Reply:
I don't really understand your point @mogoodrich .  How will it cause an issue for this to be implemented that is any worse than throwing an exception?  Presumably this method doesn't get hit at all once you add the other method that operates by id?  But at least this is a valid implementation.

Reply:
I'm 95% sure that, if it was hit, it would throw an exception with the error ""[A different object with the same identifier value was already associated with the session : [org.openmrs.module.bedmanagement.entity.Bed#1]]", as this was the originally error this method was calling.  I'd strongly prefer to explicitly throw an exception rather than implement something I'd expect to fail.  Ideally, I'd just remove this method altogether, but it's required, being an abstract method o the super class.

This is basically what the code looked like when it was failing:

https://github.com/openmrs/openmrs-module-bedmanagement/blob/519b101ab4e9de038c6a5e2e1e8f4b20040b3e11/omod/src/main/java/org/openmrs/module/bedmanagement/rest/resource/BedDetailsResource.java#L54

I believe the issue was that the Bed was getting loaded by Hibernate, but then never needed (nor passed on to the underlying service method) and when the service method loaded the Bed it failed.

Reply:
(The whole fact that this is a Bed resource that doesn't actually operate using the bed itself is a rabbit-hole I avoided going down)